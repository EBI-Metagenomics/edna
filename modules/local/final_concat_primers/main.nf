
process FINAL_CONCAT_PRIMERS {
    tag "$meta.id"
    label 'very_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
        "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"
    
    input:
    tuple val(meta), path(cat_primers)

    output:
    tuple val(meta), path("*final_concat_primers.fasta"), optional: true, emit: final_concat_primers_out
    path "versions.yml"                                 , emit: versions

    script:
    """
    cat *concat_primers.fasta > temp_concat_primers.fasta
    if [[ -s temp_concat_primers.fasta ]]; then
        awk '!a[\$0]++' temp_concat_primers.fasta > ${meta.id}_final_concat_primers.fasta
    else
       touch ${meta.id}_final_concat_primers.fasta
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: ${params.mpt_version}
    END_VERSIONS
    """

}
