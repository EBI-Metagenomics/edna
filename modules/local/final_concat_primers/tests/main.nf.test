nextflow_process {

    name "Test Process FINAL_CONCAT_PRIMERS"
    script "../main.nf"
    process "FINAL_CONCAT_PRIMERS"
    tag "modules"
    tag "final_concat_primers"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('${projectDir}/modules/local/final_concat_primers/tests/data/test_concat_primers.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.final_concat_primers_out).match("final_concat_primers_out") }
            )
        }

    }

}
