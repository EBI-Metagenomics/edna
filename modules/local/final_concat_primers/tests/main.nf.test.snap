{
    "final_concat_primers_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_final_concat_primers.fasta:md5,f26cdc22a5231d2dbd0831f1f1983dc6"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T16:07:55.845491886"
    }
}